recipes/devil/all/conanfile.py

```diff
@@ -0,0 +1,150 @@
+import os
+
+
+required_conan_version = ">=2.1"
+
+
+class DevilConan(ConanFile):
+    name = "devil"
+    description = "Developer's Image Library (DevIL) is a programmer's library to develop applications with very " \
+                  "powerful image loading capabilities, yet is easy for a developer to learn and use."
+    license = "LGPL-2.1-only"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://openil.sourceforge.net/"
+    topics = ("devil", "image")
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_png": [True, False],
+        "with_jpeg": [True, False],
+        "with_tiff": [True, False],
+        "with_jasper": [True, False],
+        "with_squish": [True, False],
+        "with_lcms": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_png": True,
+        "with_jpeg": True,
+        "with_tiff": True,
+        "with_jasper": True,
+        "with_squish": True,
+        "with_lcms": True
+    }
+
+    implements = ["auto_shared_fpic"]
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        if self.options.with_png:
+            self.requires("libpng/[>=1.6 <2]")
+        if self.options.with_jpeg:
+            self.requires("libjpeg/9e")
+        if self.options.with_tiff:
+            self.requires("libtiff/4.7.0")
+        if self.options.with_jasper:
+            self.requires("jasper/4.2.4")
+        if self.options.with_squish:
+            self.requires("libsquish/1.15")
+        if self.options.with_lcms:
+            self.requires("lcms/2.16")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+        self._patch_sources()
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+
+        # The CMakeLists will try to call find_package regardless of options
+        # ensure that even if they are picked up by the system, they are not used
+        tc.variables["IL_NO_PNG"] = not self.options.with_png
+        tc.variables["IL_NO_JPG"] = not self.options.with_jpeg
+        tc.variables["IL_NO_TIF"] = not self.options.with_tiff
+        tc.variables["IL_NO_JP2"] = not self.options.with_jasper
+        tc.variables["IL_NO_LCMS"] = not self.options.with_lcms
+        tc.variables["IL_USE_DXTC_SQUISH"] = self.options.with_squish
+
+        tc.generate()
+
+        deps = CMakeDeps(self)
+        deps.generate()
+
+    def _patch_sources(self):
+        apply_conandata_patches(self)
+        # Let Conan handle the shared/static build
+        replace_in_file(self, os.path.join(self.source_folder, "DevIL", "src-ILU", "CMakeLists.txt"),
```

Comment:
Should we patch it or just accept upstream will and provide only shared library?

Reply:
Understood, the project generate both libraries at same time.

Reply:
I think this is good as-is, letting Conan chose it on its side. If there are issues with this being static on static builds, we will gadly accept reports!